### PR TITLE
highlight: constfold.d

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Comment by UplinkCoder** [comment]
fail270.d(12) needs updating.

File: src/ddmd/constfold.d [hunk 1]
```diff
@@ -1226,7 +1226,7 @@ extern (C++) UnionExp Index(Type type, Expression e1, Expression e2)
         uinteger_t i = e2.toInteger();
         if (i >= es1.len)
         {
-            e1.error("string index %llu is out of bounds [0 .. %llu]", i, cast(ulong)es1.len);
+            e1.error("string index %llu is out of bounds `[0 .. %llu]`", i, cast(ulong)es1.len);
```
Comment (CyberShadow) [inline src/ddmd/constfold.d]: Why not highlight the index too?

Comment (WalterBright) [inline src/ddmd/constfold.d]: I thought about that, and decided against it:

1. It isn't really code
2. Numbers are quite distinct anyway
3. It's done this way throughout the error message strings

Comment (CyberShadow) [inline src/ddmd/constfold.d]: I think it would be better to highlight anything that comes from the user code, which in most cases means all %* arguments.

**Review by andralex — APPROVED** [review]

**Pull request merged**